libbeat/common/http.go

```diff
@@ -0,0 +1,31 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package common
+
+import "net/http"
+
+// CloseIdleConnections closes any idle connections if the transport allow it.
+func CloseIdleConnections(http *http.Client) {
+	type ci interface {
+		CloseIdleConnections()
+	}
+
+	if t, ok := http.Transport.(ci); ok {
+		t.CloseIdleConnections()
+	}
```

Comment:
btw, an interface cast with one method can also be written like this:

```
if t, ok := http.Transport.(interface{ CloseIdleConnections() }); ok {
  t.CloseIdleConnections()
}
```

Reply:
Nice trick I didn't know that.